tools/libs/light/libxl_pcid.c

```diff
@@ -854,6 +854,30 @@ static void server_fini_one(libxl__gc *gc, struct vchan_client *client)
     free(client);
 }
 
+static bool is_vchan_exist(libxl_ctx *ctx, char *xs_path, char *watch_dir)
+{
+    char **dir = NULL;
+    unsigned int nb, i;
+    bool ret = false;
+    GC_INIT(ctx);
+
+    dir = xs_directory(ctx->xsh, XBT_NULL, xs_path, &nb);
```

Comment:
You can call `xs_directory(ctx->xsh, XBT_NULL, watch_dir, &nb)`. It will return `NULL` if `watch_dir` does not exist.

So, you will not need `for` loop below.

Reply:
Done